<div style="text-align: center;">
  <img src="https://github.com/Hack-io-Data/Imagenes/blob/main/01-LogosHackio/logo_naranja@4x.png?raw=true" alt="esquema" />
</div>

# Laboratorio Agrupamiento de Datos

En este laboratorio usaremos el DataFrame creado en la lección anterior resultado de la unión de los conjuntos de datos compartidos contigo. 

**Instrucciones:**

1. Lee cuidadosamente el enunciado de cada ejercicio.

2. Implementa la solución en la celda de código proporcionada.

3. Devuelve los resultados siempre en formato de DataFrame

In [13]:
import pandas as pd

netflix_producciones = pd.read_csv(r'C:\Users\fatim\Hackio\Laboratorios\Pandaslab\laboratorio-Pandas-15ejerciciosgroupby\laboratorio-Pandas_15ejerciciosgroupby\data.netflix.csv')

display(netflix_producciones.info())
print(netflix_producciones.columns)



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 513 entries, 0 to 512
Data columns (total 20 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Unnamed: 0    513 non-null    int64  
 1   show_id       513 non-null    object 
 2   type          513 non-null    object 
 3   title         513 non-null    object 
 4   director      513 non-null    object 
 5   cast          513 non-null    object 
 6   country       513 non-null    object 
 7   date_added    513 non-null    object 
 8   release_year  513 non-null    int64  
 9   rating        513 non-null    object 
 10  duration      513 non-null    object 
 11  listed_in     513 non-null    object 
 12  description   513 non-null    object 
 13  Title         513 non-null    object 
 14  Genre         513 non-null    object 
 15  Premiere      513 non-null    object 
 16  Runtime       513 non-null    int64  
 17  IMDB Score    513 non-null    float64
 18  Language      513 non-null    

None

Index(['Unnamed: 0', 'show_id', 'type', 'title', 'director', 'cast', 'country',
       'date_added', 'release_year', 'rating', 'duration', 'listed_in',
       'description', 'Title', 'Genre', 'Premiere', 'Runtime', 'IMDB Score',
       'Language', 'is_original'],
      dtype='object')


#### Ejercicio 1: ¿Cuántos títulos hay por cada tipo de contenido ('Movie' o 'TV Show')?

In [4]:
type_counts = netflix_producciones['type'].value_counts()

display(type_counts)

type
Movie      337
TV Show    176
Name: count, dtype: int64

*interpretación de resultado*

Películas (Movie): 337 títulos.
Programas de TV (TV Show): 176 títulos.

#### Ejercicio 2: ¿Cuál es el promedio de puntuación de las películas agrupadas por clasificación de audiencia ('rating')? Interpreta los resultados, ¿hay algo que te llame la atención? Explica por qué. 

In [15]:
puntuación_rating = netflix_producciones.groupby('rating')['IMDB Score'].mean()
print(puntuación_rating)


rating
G        5.500000
PG       6.033333
PG-13    6.179310
R        6.326786
TV-14    6.326852
TV-G     6.418182
TV-MA    6.186164
TV-PG    5.900000
TV-Y     6.200000
TV-Y7    6.288571
Name: IMDB Score, dtype: float64


*interpretación de resultados* 
Algunas conclusiones interesantes: clasificaciones más restrictivas como R, PG-13, y TV-MA no necesariamente tienen las mejores puntuaciones. Esto puede responder a que estos títulos son más variados en género y calidad.
TV-G y TV-Y7 destacan por mantener buenas puntuaciones promedio, a pesar de estar dirigidos a públicos más jóvenes que, en líneas generales, suelen ser más exigentes.

*explicación del código*
El código agrupa los datos del df netflix_producciones por la columna rating, que representa las clasificaciones de audiencia, y calcula el promedio (usando .mean()) de la puntuación de IMDb (IMDB Score) para cada grupo.

#### Ejercicio 3: ¿Cuántos títulos hay por cada país? Ordena los resultados de mayor a menor

#### Ejercicio 4: ¿Cuál es la calificación promedio en IMDb de los títulos agrupados por tipo ('Movie' o 'TV Show')?

#### Ejercicio 5: ¿Cuántos títulos se añadieron a Netflix cada año? Para resolver este ejercicio tendrás uqe convertir la columna `date_added` a formato fecha usando el método `pd.to_datetime`. [Aquí](https://pandas.pydata.org/docs/reference/api/pandas.to_datetime.html) tienes la documentación, leeala con detenimiento antes de usar el método. 

#### Ejercicio 6: ¿Cuál es el país con más títulos lanzados en el año 2020? Para resolver este ejercicio después del gropuby os puede resultar interesante el método [`idxmax`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.idxmax.html)

#### Ejercicio 7:¿Cuál es el promedio de puntuación en IMDb de los títulos agrupados por país?

#### Ejercicio 8: ¿Cuántos títulos fueron lanzados en cada año agrupados por tipo ('Movie' o 'TV Show')?

#### Ejercicio 9: ¿Cuál es el país con la mayor cantidad de títulos de clasificación 'TV-MA'?

#### Ejercicio 10. ¿Cuál es el promedio de duración de las películas agrupadas por país?

#### Ejercicio 11: ¿Cuál es el número total de títulos por país y clasificación de audiencia ('rating')?

#### Ejercicio 12: ¿Cuántos títulos fueron lanzados antes del año 2000 agrupados por país?

#### Ejercicio 13: ¿Cuál es la puntuación promedio en IMDb de los títulos lanzados en 2021 agrupados por país?

#### Ejercicio 14: ¿Cuántos títulos fueron dirigidos por cada director? Ordena los resultados de mayor a menor

#### Ejercicio 15: ¿Cuál es el lenguaje más frecuente en los títulos lanzados en los últimos 5 años agrupado por país?